tesseract/tesseract_urdf/test/tesseract_urdf_unit.cpp

```diff
@@ -0,0 +1,2231 @@
+{
+  tinyxml2::XMLDocument xml_doc;
+  EXPECT_TRUE(xml_doc.Parse(xml_string.c_str()) == tinyxml2::XML_SUCCESS);
+
+  tinyxml2::XMLElement* element = xml_doc.FirstChildElement(element_name.c_str());
+  EXPECT_TRUE(element != nullptr);
+
+  auto status = parse(type, element, available_materials);
+  if (!(*status))
+  {
+    CONSOLE_BRIDGE_logError(status->message().c_str());
+  }
+
+  return status;
+}
+
+TEST(TesseractURDFUnit, parse_origin)
+{
+  {
+    std::string str = "<origin xyz=\"0 0 0\" rpy=\"0 0 0\"/>";
+    Eigen::Isometry3d origin;
+    auto status = runTest<Eigen::Isometry3d>(origin, str, "origin");
+    EXPECT_TRUE(*status);
+    EXPECT_TRUE(origin.isApprox(Eigen::Isometry3d::Identity(), 1e-8));
+  }
+
+  {
+    std::string str = "<origin xyz=\"0 0 0\" q=\"1 0 0 0\"/>";
+    Eigen::Isometry3d origin;
+    auto status = runTest<Eigen::Isometry3d>(origin, str, "origin");
+    EXPECT_TRUE(*status);
+    EXPECT_TRUE(origin.isApprox(Eigen::Isometry3d::Identity(), 1e-8));
+  }
+
+  {
+    std::string str = "<origin xyz=\"0 2.5 0\" rpy=\"3.14 0 -4.5\"/>";
+    Eigen::Isometry3d origin;
+    auto status = runTest<Eigen::Isometry3d>(origin, str, "origin");
+    EXPECT_TRUE(*status);
+    EXPECT_TRUE(origin.translation().isApprox(Eigen::Vector3d(0, 2.5, 0), 1e-8));
+  }
+
+  {
+    std::string str = "<origin xyz=\"0 0 0\"/>";
+    Eigen::Isometry3d origin;
+    auto status = runTest<Eigen::Isometry3d>(origin, str, "origin");
+    EXPECT_TRUE(*status);
+    EXPECT_TRUE(origin.isApprox(Eigen::Isometry3d::Identity(), 1e-8));
+  }
+
+  {
+    std::string str = "<origin rpy=\"0 0 0\"/>";
+    Eigen::Isometry3d origin;
+    auto status = runTest<Eigen::Isometry3d>(origin, str, "origin");
+    EXPECT_TRUE(*status);
+    EXPECT_TRUE(origin.isApprox(Eigen::Isometry3d::Identity(), 1e-8));
+  }
+
+  {
+    std::string str = "<origin />";
+    Eigen::Isometry3d origin;
+    auto status = runTest<Eigen::Isometry3d>(origin, str, "origin");
+    EXPECT_FALSE(*status);
+  }
+}
+
+TEST(TesseractURDFUnit, parse_box)
+{
+  {
+    std::string str = "<box size=\"1 2 3\"/>";
+    tesseract_geometry::Box::Ptr geom;
+    auto status = runTest<tesseract_geometry::Box::Ptr>(geom, str, "box");
+    EXPECT_TRUE(*status);
+
+    EXPECT_NEAR(geom->getX(), 1, 1e-8);
+    EXPECT_NEAR(geom->getY(), 2, 1e-8);
+    EXPECT_NEAR(geom->getZ(), 3, 1e-8);
+  }
+
+  {
+    std::string str = "<box size=\"1 2 a\"/>";
+    tesseract_geometry::Box::Ptr geom;
+    auto status = runTest<tesseract_geometry::Box::Ptr>(geom, str, "box");
+    EXPECT_FALSE(*status);
+  }
+
+  {
+    std::string str = "<box size=\"1 2\"/>";
+    tesseract_geometry::Box::Ptr geom;
+    auto status = runTest<tesseract_geometry::Box::Ptr>(geom, str, "box");
+    EXPECT_FALSE(*status);
+  }
+
+  {
+    std::string str = "<box />";
+    tesseract_geometry::Box::Ptr geom;
+    auto status = runTest<tesseract_geometry::Box::Ptr>(geom, str, "box");
+    EXPECT_FALSE(*status);
+  }
+}
+
```

Comment:
Right now non of the code checks if there is extra stuff. but it may be good to have in case in the future someone changes the parser to where that is not true. I will add extra attributes.